src/filters/truncate.js

```diff
@@ -3,7 +3,7 @@
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
     return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+      if (typeof str !== 'string' || str.length <= charLimit) {
```

Comment:
I think we both want to make sure the string isn't longer than limit and truncate on word boundary when we can.

The intent there is that we will not look back farther than 10 characters to find a word boundary in case we're truncating in the middle of a long unbroken string like a sha identifier or uid.